streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedSegments.java

```diff
@@ -51,4 +51,10 @@ public TimestampedSegment getOrCreateSegment(final long segmentId,
             return newSegment;
         }
     }
+
+    @Override
+    public void openExisting(final InternalProcessorContext context, final long streamTime) {
+        metricsRecorder.init(context.metrics(), context.taskId());
+        super.openExisting(context, streamTime);
+    }
 }
```

Comment:
The metrics recorder is initialised for a segmented timestamped key-value store.